File: API/domain/expense.py
from domain.user import User


class Expense:
    description = ""
    payer_user = None
    amount = 0.0

    def __init__(self, description: str, payer_user: User, amount: float):
        self.description = description
        self.payer_user = payer_user
        self.amount = amount

    def to_json(self):
        return {"description": self.description, "payer_user": self.payer_user.to_json(), "amount": self.amount}

File: APP/lib/domain/expense.dart
import 'domain.dart';

class Expense {
  String description;
  User payerUser;
  double amount = 0.0;

  Expense(this.description, this.payerUser, this.amount);

  factory Expense.fromJson(dynamic json) {
    User toUser = User.fromJson(json["payer_user"]);
    var amount = json["amount"] ?? 0;
    return Expense(json["description"] as String, toUser, amount.toDouble());
  }

  @override
  String toString() {
    String name = payerUser.name;
    double rounded = double.parse((amount).toStringAsFixed(2));
    return "$description (\$$rounded): $name";
  }
}